File: 0206-reverse-linked-list/0206-reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
         if (head == null) return null;
        ListNode temp=head.next;
        ListNode prev=head;
        while(temp!=null){

            ListNode store=temp.next;
            temp.next=prev;
             if(prev==head){
                prev.next=null;
            }
            prev=temp;
            temp=store;
            
           

        }
    return prev; 
    }
}

File: 0206-reverse-linked-list/0206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head is None:
            return head
        prev=head
        temp=head.next
        while temp is not None:
            store=temp.next
            temp.next=prev
            if prev==head:
                prev.next=None
            prev=temp
            temp=store
        return prev

        